minifi/minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/service/BootstrapFileProvider.java

```diff
@@ -0,0 +1,234 @@
+    private static final String MINIFI_STATUS_FILE_NAME = "minifi.status";
+    private static final String MINIFI_LOCK_FILE_NAME = "minifi.lock";
+    private static final String DEFAULT_CONFIG_FILE = "./conf/bootstrap.conf";
+    private static final String BOOTSTRAP_CONFIG_FILE_SYSTEM_PROPERTY_KEY = "org.apache.nifi.minifi.bootstrap.config.file";
+    private static final String MINIFI_HOME_ENV_VARIABLE_KEY = "MINIFI_HOME";
+    private static final String MINIFI_PID_DIR_PROP = "org.apache.nifi.minifi.bootstrap.config.pid.dir";
+    private static final String DEFAULT_PID_DIR = "bin";
+
+    private final File bootstrapConfigFile;
+
+    public BootstrapFileProvider(File bootstrapConfigFile) {
+        if (bootstrapConfigFile == null || !bootstrapConfigFile.exists()) {
+            throw new IllegalArgumentException("The specified bootstrap file doesn't exists: " + bootstrapConfigFile);
+        }
+        this.bootstrapConfigFile = bootstrapConfigFile;
+    }
+
+    public static File getBootstrapConfFile() {
+        File bootstrapConfigFile = Optional.ofNullable(System.getProperty(BOOTSTRAP_CONFIG_FILE_SYSTEM_PROPERTY_KEY))
+            .map(File::new)
+            .orElseGet(() -> Optional.ofNullable(System.getenv(MINIFI_HOME_ENV_VARIABLE_KEY))
+                .map(File::new)
+                .map(nifiHomeFile -> new File(nifiHomeFile, DEFAULT_CONFIG_FILE))
+                .orElseGet(() -> new File(DEFAULT_CONFIG_FILE)));
+        LOGGER.debug("Bootstrap config file: {}", bootstrapConfigFile);
+        return bootstrapConfigFile;
+    }
+
+    public File getPidFile() throws IOException {
+        return getBootstrapFile(MINIFI_PID_FILE_NAME);
+    }
+
+    public File getStatusFile() throws IOException {
+        return getBootstrapFile(MINIFI_STATUS_FILE_NAME);
+    }
+
+    public File getLockFile() throws IOException {
+        return getBootstrapFile(MINIFI_LOCK_FILE_NAME);
+    }
+
+    public File getReloadLockFile() {
+        File confDir = bootstrapConfigFile.getParentFile();
+        File nifiHome = confDir.getParentFile();
+        File bin = new File(nifiHome, "bin");
+        File reloadFile = new File(bin, "minifi.reload.lock");
+
+        LOGGER.debug("Reload File: {}", reloadFile);
+        return reloadFile;
+    }
+
+    public File getSwapFile() {
+        File confDir = bootstrapConfigFile.getParentFile();
+        File swapFile = new File(confDir, "swap.yml");
+
+        LOGGER.debug("Swap File: {}", swapFile);
+        return swapFile;
+    }
+
+    public Properties getBootstrapProperties() throws IOException {
+        if (!bootstrapConfigFile.exists()) {
+            throw new FileNotFoundException(bootstrapConfigFile.getAbsolutePath());
+        }
+
+        Properties bootstrapProperties = new Properties();
+        try (FileInputStream fis = new FileInputStream(bootstrapConfigFile)) {
+            bootstrapProperties.load(fis);
+        }
+
+        logProperties("Bootstrap", bootstrapProperties);
```

Comment:
Can any of these bootstrap properties contain sensitive values? If so, they should not be logged.